document_page_approval/tests/test_document_page_approval.py

```diff
@@ -103,41 +127,153 @@ def test_change_request_from_scratch(self):
         )
 
         self.assertEqual(chreq.state, "draft")
-        self.assertNotEqual(page.content, chreq.content)
-        self.assertNotEqual(page.approved_date, chreq.approved_date)
-        self.assertNotEqual(page.approved_uid, chreq.approved_uid)
 
-        chreq.action_to_approve()
+        # Move change request to 'to approve' state
+        chreq.with_user(self.user2).action_to_approve()
         self.assertEqual(chreq.state, "to approve")
-        self.assertNotEqual(page.content, chreq.content)
-        self.assertNotEqual(page.approved_date, chreq.approved_date)
-        self.assertNotEqual(page.approved_uid, chreq.approved_uid)
 
-        chreq.action_cancel()
+        # Cancel and return to draft
+        chreq.with_user(self.user2).action_cancel()
         self.assertEqual(chreq.state, "cancelled")
-        self.assertNotEqual(page.content, chreq.content)
-        self.assertNotEqual(page.approved_date, chreq.approved_date)
-        self.assertNotEqual(page.approved_uid, chreq.approved_uid)
 
-        chreq.action_draft()
+        chreq.with_user(self.user2).action_draft()
         self.assertEqual(chreq.state, "draft")
-        self.assertNotEqual(page.content, chreq.content)
-        self.assertNotEqual(page.approved_date, chreq.approved_date)
-        self.assertNotEqual(page.approved_uid, chreq.approved_uid)
 
-        chreq.action_approve()
+        # Approve the change request
+        chreq.with_user(self.user2).action_approve()
```

Comment:
If you want `user2` to approve it, you need it to be in “to approve” status, so you will need to do the following first

```
chreq.action_to_approve()
self.assertEqual(chreq.state, "to approve")
```